src/ripple/app/tx/impl/Transactor.cpp

```diff
@@ -448,6 +465,31 @@ Transactor::apply()
     return doApply();
 }
 
+NotTEC
+Transactor::checkTemporalTxValidity(PreclaimContext const& ctx)
+{
+    using d = NetClock::duration;
+    using tp = NetClock::time_point;
+
+    if (ctx.view.rules().enabled(featureTemporalTxValidity))
+    {
+        // We check time against the close time of the parent ledger,
+        // because that is the time that the network, as a whole,
+        // agrees on.
+        auto const pct = ctx.view.parentCloseTime();
```

Comment:
Using `parentCloseTime` makes sense. 
Many people probably already know, but I was curious about what the `parentCloseTime` looked like, so I did some measurement. I connected my local machine to the mainnet overnight, and recorded my local time when ledgers were closed and the network agreed close time. 
The network agreed time follows the pattern, 0, 1, 2, 10, 11, 12, 20, 21, 30... There are the same number of `parentCloseTime` ending with 0 and ending with 1. There are less `parentCloseTime` ending with 2, (about 22%). 
The result of ```network agreed close time - my local time of closing``` are shown in the histogram below.  
![Figure_1](https://user-images.githubusercontent.com/354723/118191356-70371080-b412-11eb-9e52-86c0874049aa.png)
It seems the differences are in the range [-3, 6]. (I had one outlier at -12 when my machine lost sync.)  

When clients assign `NotValidBefore` or `NotValidAfter`, all the practical values (in tens of seconds) should be fine to avoid false positive. (Clients should also consider the time buffer (10 seconds to `NotValidAfter`) that @ximinez suggested.

Reply:
Added a section in the docs to clarify this: https://github.com/ripple/xrpl-dev-portal/pull/1073